Engine/gfx/ali3dogl.cpp

```diff
@@ -556,41 +561,47 @@ void main()
 )EOS";
 
 
-void CreateTransparencyShader(ShaderProgram &prg)
+bool CreateTransparencyShader(ShaderProgram &prg)
 {
-  CreateShaderProgram(prg, "Transparency", default_vertex_shader_src, transparency_fragment_shader_src);
-
-  prg.MVPMatrix = glGetUniformLocation(prg.Program, "uMVPMatrix");
-  prg.TextureId = glGetUniformLocation(prg.Program, "textID");
-  prg.Alpha = glGetUniformLocation(prg.Program, "alpha");
+  if(CreateShaderProgram(prg, "Transparency", default_vertex_shader_src, transparency_fragment_shader_src)) {
```

Comment:
In regards to this, and other similar changes below: I really suggest to use the "early exit" rule, instead of putting multiple lines into an extra nesting.

That is, -
<pre>
if ( !condition ) return;
// do something on success
</pre>

That way you know that the function proceeds below only on success, and don't have to change nesting too much too.